### PR TITLE
Python 3.9 Support

### PR DESCRIPTION
**Pull request by mattp0**
This uses the AWS Lambda image for python 3.9
Node is set to 16.x because node 18.x and 20.x require glibc>=2.28 and 2.27 is present on amazon linux 2

Rather than try and install node 18.x or 20.x on amazon linux 2 we can keep what we had as the primary use I have seen is to deploy the cumulus dashboard. 

This is a resolution to a dependency layer issue ASF encounter while building on amazon linux 2023. 
Ideally once CMA can be built on python 3.12 we can update to the amazon linxu 2023 and use python 3.12 and node 20.x

### CODE REVIEW

**Comment by mattp0** [comment]
This has been tested on the NISAR upgrade and the ASF upgrade and the dependency issues are resolved.

File: Dockerfile [hunk 1]
```diff
@@ -30,27 +34,25 @@ RUN \
...
 # Add user for keygen in Makefile
 ARG USER
 RUN \
         echo "user:x:${USER}:0:root:/:/bin/bash" >> /etc/passwd
 
-# Uncommenting breaks: CIRRUS-ASF's CI/CD
-# Uncommenting fixes: `fatal: detected dubious ownership in repository at '/CIRRUS-core'
-# Uncommenting fixes: `fatal: detected dubious ownership in repository at '/CIRRUS-DAAC'
-# COPY .gitconfig /.gitconfig
+RUN python3 -m pip install boto3 setuptools
```
Comment (mattp0) [inline Dockerfile:51]: this is probably not needed as IIRC boto3 comes on the amazon images

File: Dockerfile [hunk 2]
```diff
@@ -9,16 +9,20 @@ FROM amazonlinux:2023 AS core_base
 #   * AWS CLI
 #   * Terraform
 
-ENV NODE_VERSION="20.x"
-ENV TERRAFORM_VERSION="1.9.2"
-ENV AWS_CLI_VERSION="2.17.13"
+#amazon linux 2 does not support node 18.x or node 20.x glibc=2.27 and >=2.28 is required
```
Comment (gjclark) [inline Dockerfile:12]:
```suggestion
# amazon linux 2 does not support node 18.x or node 20.x glibc=2.27 and >=2.28 is required
```

File: Dockerfile [hunk 3]
```diff
@@ -30,27 +34,25 @@ RUN \
...
 RUN \
         echo "user:x:${USER}:0:root:/:/bin/bash" >> /etc/passwd
 
-# Uncommenting breaks: CIRRUS-ASF's CI/CD
-# Uncommenting fixes: `fatal: detected dubious ownership in repository at '/CIRRUS-core'
-# Uncommenting fixes: `fatal: detected dubious ownership in repository at '/CIRRUS-DAAC'
-# COPY .gitconfig /.gitconfig
+RUN python3 -m pip install boto3 setuptools
+
+#COPY .gitconfig /.gitconfig
```
Comment (gjclark) [inline Dockerfile:53]:
```suggestion
# COPY .gitconfig /.gitconfig
```

**Review by gjclark — APPROVED** [review]

**Review by mckadesorensen — APPROVED** [review]
This looks good to me. It might be a good idea to request Mike and Chris on this since it is a change to CIRRUS-core.

**Comment by mattp0** [comment]
big question here is, does moving back to node 16.x in the build image effect any of the users ? otherwise the experience will be the same. python 3.9 on an amazon linux 2 container. https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

**Review by reweeden — APPROVED** [review]

**Pull request merged**